### ISSUE TITLE
Anyone inform me when synchronization over the Internet starts working flawlessly?

### ISSUE THREAD

**Issue by trancephorm** [orbitdb/orbitdb#1022]
Seems like the biggest and not solved yet problem is web browsers connectivity over Internet. In local networks, peers are finding themselves easily, but it's a whole another story on Internet.

That is the last breakthough OrbitDB needs to made to really become popular, but this way - there's not much use for it.

On a side-note, lot of people are losing lot of time trying to make this work (including me). It would be OK if there is a disclaimer that this is not yet working on Internet, just on local networks. https://ipfs.io/ipfs/QmUsoSkGzUQnCgzfjL549KKf29m5EMYky3Y6gQp5HptLTG/  Example here just works locally.

**Comment by LucaPanofsky** [comment]
The link you posted is working properly. 
If you open the console you will read logs like: 
```Swarm listening on /dns4/webrtc-star.discovery.libp2p.io/tcp/443/wss/p2p-webrtc-star/p2p/QmNWTr1wk...```

Browsers to browsers connectivity requires a star signaling server at the moment and it works

**Comment by trancephorm** [comment]
Tried it again, it's working now here over the Internet, why it didn't work in the first attempt, I don't really know, and I apologize for that. Anyways, it is worth mentioning in Readme.MD document that this is actually still not p2p database because it needs signalling servers to make connections between Internet peers. Only in local networks this is real p2p database.

All of that web browser connectivity thing is deeply political. By now, we should've had web browsers capable of making any type of connections (with eventual user consent), just like any other apps are capable of.

**Comment by trancephorm** [comment]
Continued testing it..... it seems that Firefox is quirky connecting. Chrome and Safari worked ok so far.

**Comment by trancephorm** [comment]
Second attempt in Firefox and it works. IDK what to think other than it's not flawless to say at least.

**Comment by LucaPanofsky** [comment]
I do not know what is the problem with this example code. It might have issues. 
However, in my apps I never had issues using my self deployed star signaling server. 

The server used in this example are "development server" and there is no guarantee they will be working 100% of the times - because of many reason. 

Finally, orbitdb is distributed database built on top ipfs and libp2p.
What you are not happy with depends on libp2p, not on orbitdb. 

You can have a look here 
https://github.com/libp2p/js-libp2p/issues/385

**Comment by anarkrypto** [comment]
IPFS node behind a [NAT newtork](https://docs.ipfs.tech/concepts/glossary/#nat) (such as a residential network router) have difficulty connecting to the rest of the nodes on the IPFS. This can cause users to experience very long wait times or a high rate of request failures.

Read more about it and some configurations that can help here:
https://docs.ipfs.tech/how-to/nat-configuration/

In general we host IPFS nodes on VPS (with correct firewall settings) or third-party services  (such as Pinata, Infura, NFT.storage and etc) when in production.

**Comment by haadcode** [comment]
Sounds like the original issue was solved. Feel free to re-open or create new ones if the issue perists.

Note that ipfs/libp2p are going through some upgrades in terms of how browser-to-browser connections are handled, and that work is not yet integrated in OrbitDB. We're working on it though!
